File: CPS/Junior and Senior CPS Weekly Contests - 61/I_I.py
def solve():
    t = int(input())
    for _ in range(t):
        n = int(input())
        found = False
        for a in range(2, int(n ** (1/3)) + 2):
            if n % a != 0:
                continue
            rem = n // a
            for b in range(a+1, int(rem ** 0.5) + 2):
                if rem % b != 0:
                    continue
                c = rem // b
                if c > b and c != a and c != b and a != b and a != c and b != c and c >= 2:
                    print("YES")
                    print(a, b, c)
                    found = True
                    break
            if found:
                break
        if not found:
            print("NO")

File: CPS/Junior and Senior CPS Weekly Contests - 61/I_I.java
import java.util.*;

public class  I_I {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        
        while (t-- > 0) {
            long n = sc.nextLong();
            long a = -1, b = -1, c = -1;
            boolean found = false;

            for (long i = 2; i * i <= n; i++) {
                if (n % i == 0) {
                    a = i;
                    break;
                }
            }

            if (a != -1) {
                long m = n / a;
                for (long i = a + 1; i * i <= m; i++) {
                    if (m % i == 0 && i != a) {
                        b = i;
                        c = m / b;
                        if (c != a && c != b && c > 1) {
                            found = true;
                        }
                        break;
                    }
                }
            }

            if (found) {
                System.out.println("YES");
                System.out.println(a + " " + b + " " + c);
            } else {
                System.out.println("NO");
            }
        }
    }
}
